pkg/platform/model/inventory.go

```diff
@@ -82,21 +118,33 @@ func IngredientWithLatestVersion(language, name string, prefix NamespacePrefix)
 		return nil, locale.NewInputError("inventory_ingredient_not_available", "The ingredient {{.V0}} is not available on the ActiveState Platform", name)
 	}
 
+	var candidates []*IngredientAndVersion
 	for _, res := range results {
 		for _, feature := range res.LatestVersion.ProvidedFeatures {
 			if feature.Feature == nil || *feature.Feature != name {
 				continue
 			}
 
-			return &IngredientAndVersion{
-				res.V1SearchIngredientsResponseIngredientsItems,
-				*res.LatestVersion.Version,
-				res.Namespace,
-			}, nil
+			candidates = append(
+				candidates,
+				&IngredientAndVersion{
+					res.V1SearchIngredientsResponseIngredientsItems,
+					*res.LatestVersion.Version,
+					res.Namespace,
+				})
 		}
 	}
+	bestMatch, err := FilterForBestIngredientMatch(candidates, name)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not retrieve a match for ingredient %s.", name)
+	}
 
-	return nil, locale.NewInputError("inventory_ingredient_no_version_available", "No versions are available for package {{.V0}} on the ActiveState Platform", name)
+	latest := &IngredientAndVersion{
+		bestMatch.V1SearchIngredientsResponseIngredientsItems,
+		*bestMatch.LatestVersion.Version,
+		bestMatch.Namespace,
+	}
+	return latest, nil
```

Comment:
Can't we just return the `bestMatch` at this point?

Reply:
yes, indeed!